Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIr.kt

```diff
@@ -0,0 +1,419 @@
+            val constructorParams: List<ConstructorParamStub> = emptyList(),
+            override val superClassInit: SuperClassInit? = null,
+            override val interfaces: List<StubType> = emptyList(),
+            override val properties: List<PropertyStub> = emptyList(),
+            override val origin: StubOrigin,
+            override val annotations: List<AnnotationStub> = emptyList(),
+            override val childrenClasses: List<ClassStub> = emptyList(),
+            override val companion: Companion? = null,
+            override val functions: List<FunctionalStub> = emptyList(),
+            override val simpleContainers: List<SimpleStubContainer> = emptyList()
+    ) : ClassStub()
+
+    class Companion(
+            override val superClassInit: SuperClassInit? = null,
+            override val interfaces: List<StubType> = emptyList(),
+            override val properties: List<PropertyStub> = emptyList(),
+            override val origin: StubOrigin = StubOrigin.None,
+            override val annotations: List<AnnotationStub> = emptyList(),
+            override val childrenClasses: List<ClassStub> = emptyList(),
+            override val functions: List<FunctionalStub> = emptyList(),
+            override val simpleContainers: List<SimpleStubContainer> = emptyList()
+    ) : ClassStub() {
+        override val companion: Companion? = null
+    }
+
+    class Enum(
+            val classifier: Classifier,
+            val entries: List<EnumEntryStub>,
+            val constructorParams: List<ConstructorParamStub> = emptyList(),
+            override val superClassInit: SuperClassInit? = null,
+            override val interfaces: List<StubType> = emptyList(),
+            override val properties: List<PropertyStub> = emptyList(),
+            override val origin: StubOrigin,
+            override val annotations: List<AnnotationStub> = emptyList(),
+            override val childrenClasses: List<ClassStub> = emptyList(),
+            override val companion: Companion?= null,
+            override val functions: List<FunctionalStub> = emptyList(),
+            override val simpleContainers: List<SimpleStubContainer> = emptyList()
+    ) : ClassStub()
+
+    override val meta: StubContainerMeta = StubContainerMeta()
+
+    override val classes: List<ClassStub>
+        get() = childrenClasses + listOfNotNull(companion)
+
+    override fun <T, R> accept(visitorAux: StubIrVisitorAux<T, R>, data: T) =
+        visitorAux.visitClass(this, data)
+
+    override val typealiases: List<TypealiasStub> = emptyList()
+}
+
+class ReceiverParameterStub(
+        val type: StubType
+)
+
+class FunctionParameterStub(
+        val name: String,
+        val type: StubType,
+        override val annotations: List<AnnotationStub> = emptyList(),
+        val isVararg: Boolean = false,
+        val origin: StubOrigin = StubOrigin.None
+) : AnnotationHolder
+
+enum class MemberStubModality {
+    OVERRIDE,
+    OPEN,
+    NONE,
```

Comment:
There is no such thing as `NONE` modality.
In source code some declarations may have default modality, but in metadata modality is always explicit.
I believe it makes sense to make this modality explicit too, but render it with respect to container's default modality.